### PR TITLE
add back in the ability to type check a schema at compiletime

### PR DESCRIPTION
**Pull request by benwilson512**

### CODE REVIEW

File: lib/absinthe/schema/rule/type_names_are_valid.ex [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  @moduledoc false
+
+  @description """
+  Types must exist if referenced.
+  """
+
+  def explanation(%{data: %{identifier: identifier, parent: parent}}) do
+    artifact_name = String.capitalize(parent.name)
+    """
+    #{artifact_name} #{inspect identifier} is not defined in your schema.
```
Comment (benwilson512) [inline lib/absinthe/schema/rule/type_names_are_valid.ex]: @bruce is this a sufficient error message?

File: test/support/schemas/prefix_schema.exs [hunk 2]
```diff
@@ -3,12 +3,12 @@ defmodule PrefixSchema do
   use Absinthe.Schema
 
   query do
-    field :foo do
+    field :foo, :integer do
```
Comment (benwilson512) [inline test/support/schemas/prefix_schema.exs]: @bruce we may want to have some additional checks in the field macro that make sure a type of _some_ kind is set. Honestly I'd just raise an exception, it's basically a function call error.

Comment (bruce) [inline test/support/schemas/prefix_schema.exs]: I ran into this _exact_ thing earlier today; silent failure to add the field. Definitely need something.

File: lib/absinthe/schema/rule/type_names_are_valid.ex (hunk 1 above)
Comment (bruce) [inline lib/absinthe/schema/rule/type_names_are_valid.ex]: :+1:

**Pull request merged**